client/src/features/action/needs/NeedFormGoods.tsx

```diff
@@ -55,8 +57,13 @@ const initialFormData: ShareANeedData = {
 
 function NeedForm() {
   const [formData, setFormData] = React.useState<ShareANeedData>(initialFormData);
+  const [showPrompt, setShowPrompt] = React.useState(false);
   const history = useHistory();
 
+  React.useEffect(() => {
+    setShowPrompt(() => DetectFormData(formData));
+  }, [showPrompt, formData]);
```

Comment:
don't think we want `showPrompt` in the dependencies here
that could lead to infinite loop

Reply:
updated